### ISSUE TITLE
Running LXMERT on a custom dataset

### ISSUE THREAD

**Issue by arjun-mani** [facebookresearch/mmf#807]
Hello,

I'm trying to apply LXMERT to a custom dataset that I've created, but the loss is stuck and not moving anywhere. Here's the main changes I've made to the config file:

`model_config:
  lxmert:
    training_head_type: classification
    num_labels: 2
    losses:
    - type: logit_bce`
    
`dataset_config:
  visual7w:
    return_features_info: true
    processors:
      text_processor:
        type: bert_tokenizer
        params:
          tokenizer_config:
            type: bert-base-uncased
            params:
              do_lower_case: true
          mask_probability: 0
          max_seq_length: 20
      transformer_bbox_processor:
        type: transformer_bbox
        params:
          bbox_key: bbox
          image_width_key: image_width
          image_height_key: image_height`

I've observed the following: Pythia converges on the dataset I've created, even though LXMERT does not. In addition, LXMERT seems to converge on VQA 2.0 but on my dataset. As such my question is - what are the steps necessary to make LXMERT work on my custom dataset? I have edited the config file and believe my builder/dataset files are correct, but I'm wondering whether I have to edit the LXMERT model config file, or do something of this sort. I apologize for the general question - it's just unclear to me why this doesn't work out-of-the-box, and I'm wondering whether I'm missing something (e.g. some relevant config is hard-coded to VQA 2.0 parameters). Thanks.

**Comment by ytsheng** [comment]
Hi @arjun-mani, thanks for using mmf!

Do you mind sharing your **complete** model and dataset config?

**Comment by arjun-mani** [comment]
@ytsheng No problem. Here's my project config file:

<img width="399" alt="Screen Shot 2021-03-10 at 2 09 03 PM" src="https://user-images.githubusercontent.com/4947338/110683588-30517400-81aa-11eb-9807-1b04c249ad59.png">

Here's my dataset config file:

<img width="632" alt="Screen Shot 2021-03-10 at 2 10 43 PM" src="https://user-images.githubusercontent.com/4947338/110683785-6a227a80-81aa-11eb-8df5-8dc0623ce54c.png">

And I didn't modify the model config file for LXMERT, so it's just the one here: https://github.com/facebookresearch/mmf/blob/master/mmf/configs/models/lxmert/defaults.yaml

Thanks for the help.

**Comment by vedanuj** [comment]
Hi @arjun-mani 

There can be various factors affecting performance and convergence. Since this is a new dataset you are trying, you can try some hyperparameter sweeps to find a good set of parameters. One thing you can try is use some hyper parameters that others have found to be good for this datasets on similar transformer architectures. It might also make sense to look at the training and loss curves if you plot them in tensorboard.

**Comment by arjun-mani** [comment]
Hi @vedanuj - thanks for the reply. That's exactly right, I was able to get it training after a lot of tweaking of the learning rate schedule. Closing this issue.
